pkg/executor/point_get.go

```diff
@@ -387,19 +389,122 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
 		}
 		return nil
 	}
-	err = DecodeRowValToChunk(e.BaseExecutor.Ctx(), e.Schema(), e.tblInfo, e.handle, val, req, e.rowDecoder)
+
+	sctx := e.BaseExecutor.Ctx()
+	schema := e.Schema()
+	err = DecodeRowValToChunk(sctx, schema, e.tblInfo, e.handle, val, req, e.rowDecoder)
+	if err != nil {
+		return err
+	}
+
+	err = fillRowChecksum(sctx, 0, 1, schema, e.tblInfo, [][]byte{val}, []kv.Handle{e.handle}, req, nil)
 	if err != nil {
 		return err
 	}
 
 	err = table.FillVirtualColumnValue(e.virtualColumnRetFieldTypes, e.virtualColumnIndex,
-		e.Schema().Columns, e.columns, e.Ctx().GetExprCtx(), req)
+		schema.Columns, e.columns, sctx.GetExprCtx(), req)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func shouldFillRowChecksum(schema *expression.Schema) (int, bool) {
+	for idx, col := range schema.Columns {
+		if col.ID == model.ExtraRowChecksumID {
+			return idx, true
+		}
+	}
+	return 0, false
+}
+
+func fillRowChecksum(
+	sctx sessionctx.Context,
+	start, end int,
+	schema *expression.Schema, tblInfo *model.TableInfo,
+	values [][]byte, handles []kv.Handle,
+	req *chunk.Chunk, buf []byte,
+) error {
+	checksumColumnIndex, ok := shouldFillRowChecksum(schema)
+	if !ok {
+		return nil
+	}
+
+	var handleColIDs []int64
+	if tblInfo.PKIsHandle {
+		colInfo := tblInfo.GetPkColInfo()
+		handleColIDs = []int64{colInfo.ID}
+	} else if tblInfo.IsCommonHandle {
+		pkIdx := tables.FindPrimaryIndex(tblInfo)
+		for _, col := range pkIdx.Columns {
+			colInfo := tblInfo.Columns[col.Offset]
+			handleColIDs = append(handleColIDs, colInfo.ID)
+		}
+	}
+
+	columnFt := make(map[int64]*types.FieldType)
+	for idx := range tblInfo.Columns {
+		col := tblInfo.Columns[idx]
+		columnFt[col.ID] = &col.FieldType
+	}
+	tz := sctx.GetSessionVars().TimeZone
+	ft := []*types.FieldType{schema.Columns[checksumColumnIndex].GetType()}
+	checksumCols := chunk.NewChunkWithCapacity(ft, req.Capacity())
+	for i := start; i < end; i++ {
+		handle, val := handles[i], values[i]
+		if !rowcodec.IsNewFormat(val) {
+			checksumCols.AppendNull(0)
+			continue
+		}
```

Comment:
Will this be explained in user documentation?

Reply:
no, there is no changes compare to the previous implementation, just fill the null as placeholder, and reset by the `fillChecksum` method.

Reply:
I got your point, I think we can update the documentation to let user's know tat checksum only calculated on the new format.